### PR TITLE
feat: add api to remove private key

### PR DESCRIPTION
**Pull request by MishkaRogachev**
For https://github.com/status-im/status-desktop/issues/11475

Important changes:
- [x] API for removing community private key
- [x] Separate `IsOwner` from `HasPrivateKey`
- [x] Remove Private key from syncing

### CODE REVIEW

**Review by jrainville — COMMENTED** [review]
Looks good so far

File: protocol/communities/manager.go [hunk 1]
```diff
@@ -607,13 +607,13 @@ func (m *Manager) CreateCommunityTokenPermission(request *requests.CreateCommuni
 		return nil, nil, err
 	}
 
-	if community.IsOwner() {
+	if community.IsControlNode() {
```
Comment (0x-r4bbit) [inline protocol/communities/manager.go]: For all of these where we publish a community description, I think we still want to just check for `isOwner()` no?

Comment (MishkaRogachev) [reply protocol/communities/manager.go]: fixed

File: protocol/communities/community.go [hunk 2]
```diff
@@ -233,6 +237,10 @@ func (o *Community) MarshalJSON() ([]byte, error) {
 	}{
 		ID:                          o.ID(),
 		MemberRole:                  o.MemberRole(o.MemberIdentity()),
+		IsAdmin:                     o.IsAdmin(),
+		IsOwner:                     o.IsOwner(),
+		IsControlNode:               o.IsControlNode(),
+		HasAdminPrivileges:          o.IsOwnerOrAdmin(),
```
Comment (osmaczko) [inline protocol/communities/community.go]: The properties `IsAdmin` and `IsOwner` are mutually exclusive and are already better(imo) represented by the `MemberRole`.

Comment (osmaczko) [inline protocol/communities/community.go]: I am wondering if we really need to expose that, can't clients simply check if member role is either owner or admin? I wouldn't consider it as crucial logic duplication among clients.

File: protocol/protobuf/pairing.proto [hunk 3]
```diff
@@ -138,7 +138,8 @@ message SyncInstallationPublicChat {
 message SyncCommunity {
   uint64 clock = 1;
   bytes id = 2;
-  bytes private_key = 3;
+  // Don't sync private_key vecause we want to have only one control node
+  bytes private_key = 3 [deprecated = true];
```
Comment (osmaczko) [inline protocol/protobuf/pairing.proto:142]: :+1:

File: protocol/communities/community.go (hunk 2 above)
Comment (mprakhov) [reply protocol/communities/community.go]: +1, MemberRole will be enough
P.S. IsControlNode is ok

File: protocol/protobuf/pairing.proto [hunk 4]
```diff
@@ -138,7 +138,8 @@ message SyncInstallationPublicChat {
 message SyncCommunity {
   uint64 clock = 1;
   bytes id = 2;
-  bytes private_key = 3;
+  // Don't sync private_key vecause we want to have only one control node
```
Comment (jrainville) [inline protocol/protobuf/pairing.proto]:
```suggestion
  // Don't sync private_key because we want to have only one control node
```

**Review by jrainville — APPROVED** [review]
LGTM

**Review by mprakhov — APPROVED** [review]
LGTM, all other stuff I'll fix in scope of my task where owner without privateKey must behave like an admin
For this purposes I'll need your MR :)

File: protocol/communities/community.go (hunk 2 above)
Comment (MishkaRogachev) [reply protocol/communities/community.go]: returned back!

Comment (MishkaRogachev) [reply protocol/communities/community.go]: fixed

**Review by osmaczko — APPROVED** [review]

**Pull request merged**